src/Quadrature.cpp

```diff
@@ -0,0 +1,1305 @@
+    log_printf(ERROR, "Unable to set phi for azim = %d to %f which is not "
+               "in the range (0.0, PI/2)", azim, phi);
+
+  else if (azim >= _num_azim/4)
+    log_printf(ERROR, "Unable to set phi for azim = %d since azim is not in"
+               " the range (0, _num_azim/4)", azim);
+
+  else if (_phis == NULL)
+    _phis = new double[_num_azim/2];
+
+  _phis[azim] = phi;
+  _phis[_num_azim/2 - azim - 1] = M_PI - phi;
+}
+
+
+/**
+ * @brief Sets the azimuthal spacing for the given index
+ * @param spacing the spacing in the azimuthal direction to be set
+ * @param azim the azimuthal index
+ */
+void Quadrature::setAzimSpacing(FP_PRECISION spacing, int azim) {
+
+  if (spacing <= 0.0)
+    log_printf(ERROR, "Unable to set azimuthal spacing for azim = %d to %f "
+                      "which is not strictly greater than zero", azim,
+                      spacing);
+
+  else if (azim >= _num_azim/4)
+    log_printf(ERROR, "Unable to set azimuthal spacing for azim = %d since "
+                      " azim is not in the range (0, _num_azim/4)", azim);
+
+  else if (_azim_spacings == NULL)
+    _azim_spacings = new FP_PRECISION[_num_azim/2];
+
+  setAzimuthalValues(_azim_spacings, azim, spacing);
+}
+
+
+/**
+ * @brief Sets the azimuthal weight for the given index
+ * @param weight the weight of the azimuthal angle
+ * @param azim the azimuthal index
+ */
+void Quadrature::setAzimWeight(double weight, int azim) {
+
+  if (weight <= 0.0 || weight >= M_PI_2)
+    log_printf(ERROR, "Unable to set azim weight for azim = %d to %f which is "
+               "not in the range (0.0, PI/2)", azim, weight);
+
+  else if (azim >= _num_azim/4)
+    log_printf(ERROR, "Unable to set azim weight for azim = %d since azim is "
+               "not in the range (0, _num_azim/4)", azim);
+
+  else if (_azim_weights == NULL)
+    _azim_weights = new FP_PRECISION[_num_azim/2];
+
+  setAzimuthalValues(_azim_weights, azim, FP_PRECISION(weight));
+}
+
+
+/**
+ * @brief Sets the polar weight for the given indexes
+ * @param weight the weight of the polar angle
+ * @param azim the azimuthal index corresponding to the angle
+ * @param azim the polar index corresponding to the angle
+ */
+void Quadrature::setPolarWeight(FP_PRECISION weight, int azim, int polar) {
+
+  if (weight <= 0.0 || weight >= M_PI_2)
+    log_printf(ERROR, "Unable to set polar weight for azim = %d and "
+               "polar = %d to %f which is not in the range (0.0, PI/2)",
+               azim, polar, weight);
+
+  else if (azim >= _num_azim/4)
+    log_printf(ERROR, "Unable to set polar weight for azim = %d and polar = %d "
+               "since azim is not in the range (0, _num_azim/4)", azim, polar);
+
+  else if (polar >= _num_polar/2)
+    log_printf(ERROR, "Unable to set polar weight for azim = %d and polar = %d "
+               "since polar is not in the range (0, _num_polar/2)", \
+               azim, polar);
+
+  else if (_polar_weights == NULL) {
```

Comment:
Although it doesn't actually matter here, I think it is better to make this memory allocation block a standalone `if` rather than an `else if` since it doesn't logically connect to the error checking blocks above.